ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java

```diff
@@ -295,6 +295,7 @@ public void onCollectExtraUpdates(UIViewOperationQueue uiViewOperationQueue) {
           newRight == mAbsoluteRight &&
```

Comment:
Thanks for the repro: I just looked through the code here and this check is just not correct. It should all be handled by hasNewLayout above. As such, the correct fix is to just combine lines 301-304 into lines 289-292 and delete this entire if-block.